modules/store/spec/action_with_modifiers.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { ActionCreator, TypedAction } from '../src/models';
+import { createActionWithModifiers } from '../src/action_with_modifiers_creator';
+import { props } from '../src/action_creator';
+
+describe('createActionWithModifiers', () => {
+  it('should inject properties onto action', () => {
+    // given
+    const notificationMessageKey = 'notification-message';
+    const hasNotification = <P>(config: { message: string }) => <
+      T extends string
+    >(
+      action: ActionCreator<T, (p: P) => TypedAction<T> & P>
+    ) => {
+      const creator = (p: P) => ({
+        ...action(p),
+        [notificationMessageKey]: config.message,
+      });
+      creator.type = action.type;
+      return creator;
+    };
```

Comment:
Ideally there should be a function which creates modifiers for you such that you could do:

```typescript
const hasNotification = createActionModifier(
  (config: { message: string }) => ({ [notificationMessageKey]: config.message })
);
```